### PR TITLE
[QT-304] Add enos ui scenario

### PR DESCRIPTION
**Pull request by mikebaum**
Added a new enos scenario that can run the UI tests. The scenario supports two variants: edition (oss or ent) and backend (raft or consul). Tests can be run in a completely automated manner or semi-automated when trouble shooting or adding new UI tests. The details of how to run the scenario and all the supported options is documented in the enos README at: ./enos/README.md.

### CODE REVIEW

**Review by joshbrand — APPROVED** [review]
looks fantastic!

File: .github/workflows/enos-test-ui.yml
Comment (ryancragun) [inline .github/workflows/enos-test-ui.yml:2]: Similar to how we're renaming in the replication PR, would you mind updating the workflow and name to lead with `test`, eg: `test-enos-scenario-ui`?

Comment (ryancragun) [reply .github/workflows/enos-test-ui.yml:2]: The idea being here that not everybody knows what enos is, but they probably have a good idea what `test` means.

And since we're adding a lot of these it would be nice to have a common prefix.

File: .github/workflows/enos-test-ui.yml
Comment (ryancragun) [inline .github/workflows/enos-test-ui.yml:9]: This is 🥇

File: .github/workflows/enos-test-ui.yml
Comment (ryancragun) [inline .github/workflows/enos-test-ui.yml:5]: Shouldn't this be `workflow_dispatch` since we want to run it from the Actions tab?

File: .github/workflows/enos-test-ui.yml
Comment (ryancragun) [inline .github/workflows/enos-test-ui.yml:14]: Just cause I know you'll like this. If we change this to `workflow_dispatch` we can actually give people `choice` types!

```yaml
    inputs:
      storage_backend:
        description: The Vault storage backend
        required: true
        default: raft 
        type: choice
        options:
        - raft
        - consul
```

File: .github/workflows/enos-test-ui.yml
Comment (ryancragun) [inline .github/workflows/enos-test-ui.yml:19]: Ooooo

File: .github/workflows/enos-test-ui.yml
Comment (ryancragun) [inline .github/workflows/enos-test-ui.yml:30]: There are two things I've though of about this.
* Lets add a comment since we know that there's funky stuff that happens if you're not using the correct version of node.
* Why does the ui  `.nvmrc` still have 14? It would be dope to get the actual version from that file or some other source of truth that I might be unaware of.

File: .github/workflows/enos-test-ui.yml
Comment (ryancragun) [inline .github/workflows/enos-test-ui.yml:33]: nit: there are newer major versions of `actions/checkout` and some other actions you've included here, should we update those?

File: .github/workflows/enos-test-ui.yml
Comment (ryancragun) [inline .github/workflows/enos-test-ui.yml:64]: I believe `@v3` gets rid of deprecation warnings.

File: .github/workflows/enos-test-ui.yml
Comment (ryancragun) [inline .github/workflows/enos-test-ui.yml:98]: We can remove this comment as 0.0.16 is what you'll get by default now. Whether or not deleting should fail the job is a different story I guess. I think it's fine to let it proceed on failure as we have the reaper workflow in place.

File: enos/enos-scenario-autopilot.hcl
Comment (ryancragun) [inline enos/enos-scenario-autopilot.hcl:49]: I feel like Jaymala did something similar in an older PR and it had weird effects. Have you tried this to ensure stuff behaves as we expect?

File: ui/scripts/enos-test-ember.js
Comment (ryancragun) [inline ui/scripts/enos-test-ember.js:49]: nice improvement

**Review by ryancragun — COMMENTED** [review]
Looking good!

File: .github/workflows/enos-test-ui.yml
Comment (ryancragun) [reply .github/workflows/enos-test-ui.yml:5]: I guess maybe we want both, since we'll call this from the nightly but probably want to make executing it super easy for a branch.

Comment (mikebaum) [reply .github/workflows/enos-test-ui.yml:5]: Yeah I like that idea... done.

File: .github/workflows/enos-test-ui.yml
Comment (mikebaum) [reply .github/workflows/enos-test-ui.yml:14]: awesome, I like it.

File: ui/scripts/enos-test-ember.js
Comment (mikebaum) [reply ui/scripts/enos-test-ember.js:49]: yeah, I learned that it's not possible to add more than one filter.

File: .github/workflows/enos-test-ui.yml
Comment (mikebaum) [reply .github/workflows/enos-test-ui.yml]: good idea, updated all the actions to the latest versions for those that had newer versions.

File: .github/workflows/enos-test-ui.yml
Comment (mikebaum) [reply .github/workflows/enos-test-ui.yml]: done.

File: .github/workflows/enos-test-ui.yml
Comment (mikebaum) [reply .github/workflows/enos-test-ui.yml]: updated it to be dynamic... great suggestion.

File: .github/workflows/test-enos-scenario-ui.yml
Comment (jaymalasinha) [inline .github/workflows/test-enos-scenario-ui.yml:31]: note to be removed before merge

Comment (mikebaum) [reply .github/workflows/test-enos-scenario-ui.yml:31]: will do..., just added it recently to test the changes I made per the code review

**Review by jaymalasinha — COMMENTED** [review]
Great job porting UI tests over to Enos 👍 ! Just left a couple minor comments.

File: enos/enos-scenario-autopilot.hcl
Comment (jaymalasinha) [reply enos/enos-scenario-autopilot.hcl:49]: I remember adding `try` having unexpected effects, couldn't dig out the exact change though.  But if your local test breaks may be can try setting `vault_product_version` as a local variable and use that.

Comment (mikebaum) [reply enos/enos-scenario-autopilot.hcl:49]: So I tested the autopilot scenario on vault-enterprise, you can see it here: https://github.com/hashicorp/vault-enterprise/actions/runs/3968445250/jobs/6802027552. It seems to work fine, so I think these changes are okay.

**Review by manthonygfp — APPROVED** [review]
lgtm as long as you remember to remove that push trigger to your branch like you and Jaymala talked about (I only bring this up as this was a request for a final review).

**Comment by mikebaum** [comment]
> lgtm as long as you remember to remove that push trigger to your branch like you and Jaymala talked about (I only bring this up as this was a request for a final review).

man thanks for reminding me!

File: enos/enos-scenario-autopilot.hcl
Comment (jaymalasinha) [reply enos/enos-scenario-autopilot.hcl:49]: This condition was recently removed from the autopilot scenario, so we don't need to update this file here anymore.

Comment (mikebaum) [reply enos/enos-scenario-autopilot.hcl]: yeah it was indeed removed.

File: .github/workflows/test-enos-scenario-ui.yml
Comment (mikebaum) [reply .github/workflows/test-enos-scenario-ui.yml]: not currently

File: .github/workflows/test-enos-scenario-ui.yml
Comment (jaymalasinha) [inline .github/workflows/test-enos-scenario-ui.yml:101]:
```suggestion
      - name: Set Up Chrome
```

File: .github/workflows/test-enos-scenario-ui.yml
Comment (jaymalasinha) [inline .github/workflows/test-enos-scenario-ui.yml:135]: we don't need the continue-on-error here anymore as the related bug was fixed in enos

**Review by jaymalasinha — APPROVED** [review]
LGTM 👍! Left couple minor review comments but nothing blocking

**Review by ryancragun — APPROVED** [review]

File: .github/workflows/test-enos-scenario-ui.yml
Comment (mikebaum) [reply .github/workflows/test-enos-scenario-ui.yml:121]: Done 😺

File: .github/workflows/test-enos-scenario-ui.yml
Comment (mikebaum) [reply .github/workflows/test-enos-scenario-ui.yml:134]: Yeah, I like this approach. For this scenario it was necessary though, since if the only reason the scenario fails is due to the UI tests failing, we would not want to re-run it.

**Pull request merged**
